kernel/src/arch/x86_64/rackscale/get_shmem_frames.rs

```diff
@@ -148,8 +147,7 @@ pub(crate) fn handle_get_shmem_frames(
             log::debug!("Received node assignment from DCM: node {:?}", mid);
 
             // TODO(error_handling): should handle errors gracefully here, maybe percolate to client?
-            let mut shmem_managers = SHMEM_MEMSLICE_ALLOCATORS.lock();
-            let mut manager = &mut shmem_managers[mid - 1];
+            let mut manager = &mut SHMEM_MEMSLICE_ALLOCATORS[mid - 1].lock();
```

Comment:
I don't know enough about `dcm_resouce_alloc` but I think if e.g., mids[] is not sorted, this function could deadlock if two threads execute it at the same time?

Reply:
I think you're right, I misread and the lock is only alive for the loop iteration so it shouldn't cause a deadlock.